sourced/ml/cmd_entries/repos2bow.py

```diff
@@ -6,42 +6,46 @@
 from sourced.ml.extractors import __extractors__
 from sourced.ml.models import BOW, OrderedDocumentFrequencies, DocumentFrequencies
 from sourced.ml.transformers import Engine, UastExtractor, UastDeserializer, Uast2DocFreq, \
-    Uast2TermFreq, HeadFiles, TFIDF
-from sourced.ml.utils import create_engine
+    Uast2TermFreq, HeadFiles, TFIDF, Cacher, Indexer
+from sourced.ml.utils import create_engine, EnineConstants
 
 
 def repos2bow_entry(args):
     log = logging.getLogger("repos2bow")
     engine = create_engine("repos2bow-%s" % uuid4(), **args.__dict__)
-    document_column_name = "repository_id"
+    document_column_name = EnineConstants.REPOSITORY_COLUMN_NAME
     extractors = [__extractors__[s](
         args.min_docfreq, **__extractors__[s].get_kwargs_fromcmdline(args))
         for s in args.feature]
 
     uast_pipeline = Engine(engine, explain=args.explain) \
         .link(HeadFiles()) \
         .link(UastExtractor(languages=args.languages)) \
+        .link(Cacher.maybe(args.persist)) \
         .link(UastDeserializer())
     df_pipeline = uast_pipeline.link(Uast2DocFreq(extractors, document_column_name))
     df = df_pipeline.execute()
     tf_pipeline = uast_pipeline.link(Uast2TermFreq(extractors, document_column_name))
     tf = tf_pipeline.execute()
-    tfidf = TFIDF(tf=tf, df=df).execute()
 
-    tokens, documents, values = zip(*tfidf.take(tfidf.count()))
-    log.info("Creating index mappings")
-    doc2id = {doc: doc_id for doc_id, doc in enumerate(sorted(list(set(documents))))}
-    token2id = {token: token_id for token_id, token in enumerate(sorted(list(set(tokens))))}
+    document_idexer = Indexer("document")
+    token_indexer = Indexer("token")
```

Comment:
Should be EngineConstants too

Reply:
it is not from the engine. It is from `tdidf` Transformer. I create Row names on my own.

Reply:
OK, so place them in your class then